include/deal.II/grid/tria.h

```diff
@@ -3359,6 +3359,18 @@ class Triangulation : public Subscriptor
     std::pair<std::pair<cell_iterator, unsigned int>, std::bitset<3>>> &
   get_periodic_face_map() const;
 
+  /**
+   * TODO
+   */
+  const std::vector<ReferenceCell::Type> &
+  get_reference_cell_types() const;
+
+  /**
+   * TODO
+   */
+  bool
+  all_reference_cell_types_are_hyper_cube() const;
```

Comment:
@jppelteret What do you think about this function? Is this what you had in mind at our last chat?

Reply:
Yes, precisely @peterrum. Thanks for adding it! I would want to use it for much the same purposes as you've done [here](https://github.com/dealii/dealii/pull/11140/files#diff-a3f9b6c7a51a53e5c8dc202a4dc8b8ab87f574023fb5f04dd3d2d87ecc62d68bR9814-R9818).

Reply:
Thanks for the feedback! It took some time until I had time to work in this issue ;)